### PR TITLE
Add a `kube_priority_class` label to metrics coming from pods with a priority class

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Add a `kube_priority_class` label to metrics coming from pods with a [priority class](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority).

### Motivation

Allow users to filter their pod and container metrics with the pod priority class.

### Describe how to test your changes

Add a [PriorityClass](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption) to a monitored pod and check that the metrics generated from this pod have the new `kube_priority_class` label.

### Checklist

- [X] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [X] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [X] The appropriate `team/..` label has been applied, if known.
- [X] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [X] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by clamoriniere — APPROVED** [review]

File: releasenotes/notes/Add-kube_priority_class-label-62967efb50864d66.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    On Kubernetes, add a `kube_priority_class` label on metrics coming from pods with a priority class.
```
Comment (mfpierre) [inline releasenotes/notes/Add-kube_priority_class-label-62967efb50864d66.yaml:11]:
```suggestion
    On Kubernetes, add a `kube_priority_class` tag on metrics coming from pods with a priority class.
```

File: pkg/tagger/collectors/kubelet_extract.go [hunk 2]
```diff
@@ -85,6 +85,9 @@ func (c *KubeletCollector) parsePods(pods []*kubelet.Pod) ([]*TagInfo, error) {
 		// Pod phase
 		tags.AddLow("pod_phase", strings.ToLower(pod.Status.Phase))
 
+		// Priority class
+		tags.AddLow("kube_priority_class", pod.Spec.PriorityClassName)
```
Comment (mfpierre) [inline pkg/tagger/collectors/kubelet_extract.go:89]: we can probably edit a unit test case to test extracting this field

**Review by mfpierre — APPROVED** [review]
LGTM overall

File: pkg/tagger/collectors/kubelet_extract_test.go [hunk 3]
```diff
@@ -325,6 +335,54 @@ func TestParsePods(t *testing.T) {
 				StandardTags: []string{},
 			}},
 		},
+		{
+			desc: "pod with priority class + k8s recommended tags",
+			pod: &kubelet.Pod{
+				Metadata: kubelet.PodMetadata{
+					Name:      "dd-agent-rc-qd876",
+					Namespace: "default",
+					Labels: map[string]string{
```
Comment (davidor) [inline pkg/tagger/collectors/kubelet_extract_test.go:344]: Are these labels relevant to the test? If not, would it make sense to delete them to simplify?

**Review by davidor — APPROVED** [review]

File: pkg/tagger/collectors/kubelet_extract_test.go (hunk 3 above)
Comment (L3n41c) [reply pkg/tagger/collectors/kubelet_extract_test.go]: Indeed !
That would simplify the test. Done !

File: pkg/tagger/collectors/kubelet_extract.go (hunk 2 above)
Comment (L3n41c) [reply pkg/tagger/collectors/kubelet_extract.go:89]: Test case added.

File: releasenotes/notes/Add-kube_priority_class-label-62967efb50864d66.yaml (hunk 1 above)
Comment (L3n41c) [reply releasenotes/notes/Add-kube_priority_class-label-62967efb50864d66.yaml]: Fixed.
